contrib/extractor.sh

```diff
@@ -0,0 +1,184 @@
+#!/bin/bash
+
+# Set client and server directories as environment variables
+# if you don't want to enter them here
+if [ -z $CLIENT_DIR ]; then
+echo -n "Enter directory where client files are located: "
+read CLIENT_DIR
+
+if ! cd $CLIENT_DIR; then
+        echo "Cannot access $CLIENT_DIR"
+        exit 1
+fi
+if [ ! -d $CLIENT_DIR/Data ]; then
+        echo "Cannot find client files in $CLIENT_DIR"
+        exit 1
+fi
+fi
+
+if [ -z $SERVER_DIR ]; then
+echo -n "Enter directory where server files are located: "
+read SERVER_DIR
+
+if ! cd $SERVER_DIR; then
+        echo "Cannot access $SERVER_DIR"
+        exit 1
+fi
+fi
+
+# These should all be correct on a standard installation
+DATA_DIR=${SERVER_DIR}/data
+MAPEXTRACTOR=${SERVER_DIR}/bin/mapextractor
+VMAP4EXTRACTOR=${SERVER_DIR}/bin/vmap4extractor
+VMAP4ASSEMBLER=${SERVER_DIR}/bin/vmap4assembler
+MMAPS_GENERATOR=${SERVER_DIR}/bin/mmaps_generator
+
+check_files() {
+for i in $MAPEXTRACTOR $VMAP4EXTRACTOR $VMAP4ASSEMBLER $MMAPS_GENERATOR
+do
+	if [[ ! -x $i ]]
+	then
+		echo "Invalid executable: $i"
+		echo "Check tools are installed correctly"
```

Comment:
suggestion (for better understanding & readability in proper English):
```suggestion
		echo "Check that the extraction tools are installed correctly"
```

Reply:
This is normal shell script error language, it looks a bit weird if you put errors into full normal English :)